### PR TITLE
Bower follow-ups

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: plugins/package-managers/bower/src/main/kotlin/Bower.kt
Comment (sschuberth) [inline plugins/package-managers/bower/src/main/kotlin/Bower.kt:197]: Commit message: Typo in "funcation".

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: plugins/package-managers/bower/src/main/kotlin/Bower.kt
Comment (sschuberth) [inline plugins/package-managers/bower/src/main/kotlin/Bower.kt:193]: As you have `with(pkgMeta)` above, I believe this could be simplified to just

    "$name:$version".takeUnless { name == null || version == null }

File: plugins/package-managers/bower/src/main/kotlin/Bower.kt
Comment (sschuberth) [inline plugins/package-managers/bower/src/main/kotlin/Bower.kt]: This triggers an inspection hint about the use of `!!`. I wonder whether we can work around that by making `key` non-nullable, e.g. by ensuring to not call it on root projects?

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: plugins/package-managers/bower/src/main/kotlin/Bower.kt
Comment (fviernau) [reply plugins/package-managers/bower/src/main/kotlin/Bower.kt]: Is it just me or have the warnings about `!!` become a bit too much. 

I just wanted to make obvious what previously was a bit hidden, but this warning blocks the change right?
(If so, is it possible to go back to the previous way of flagging `!!` ?)

Comment (sschuberth) [reply plugins/package-managers/bower/src/main/kotlin/Bower.kt]: Yes, this blocks the merge. But previously `!!` was not flagged at all, I believe. I prefer the current way of flagging it, and if you're *really* sure `!!` is the right thing to do in a specific case, it should be whitelisted via `@Suppress("UnsafeCallOnNullableType")`.

File: plugins/package-managers/bower/src/main/kotlin/Bower.kt
Comment (fviernau) [reply plugins/package-managers/bower/src/main/kotlin/Bower.kt:193]: Would this be a change in behavior in case of of them is `""` ?

File: plugins/package-managers/bower/src/main/kotlin/Bower.kt
Comment (fviernau) [reply plugins/package-managers/bower/src/main/kotlin/Bower.kt]: Can we revisit the decision whether we want to flag all uses of `!!` really? 
I've spent quite some time working around cases, where `!!` was perfectly fine. Huge waste of time imo. 
If there is a convincing rationale why `!!` should be forbidden, I may change my mind. But I'm not aware of any, in particular
1. `!!` seems forbidden while `checkNotNull()` seems fine
2. Whether `!!` is ok is a case by case decision, and I'm pretty sure all approvers in our community already 
    pay good attention to any ocurence of `!!`, so flagging is not needed as we handle it already.

Comment (fviernau) [reply plugins/package-managers/bower/src/main/kotlin/Bower.kt]: I went for the `supress`, because the observation that the previous `checkNotNull` never failed implies that no lockup with a `null` key has been done.

File: plugins/package-managers/bower/src/main/kotlin/Bower.kt
Comment (fviernau) [reply plugins/package-managers/bower/src/main/kotlin/Bower.kt]: Anyhow, good catch with the `with(pkgMeta)`. This was a left over which I dropped now.

Comment (sschuberth) [reply plugins/package-managers/bower/src/main/kotlin/Bower.kt]: > Would this be a change in behavior in case of of them is `""` ?

Yes, empty names / versions would not be discarded anymore. Not sure if this would actually be any issue though.

File: plugins/package-managers/bower/src/main/kotlin/Bower.kt
Comment (sschuberth) [reply plugins/package-managers/bower/src/main/kotlin/Bower.kt]: > Can we revisit the decision whether we want to flag all uses of `!!` really?

Sure, feel free to bring it up in a Kotlin dev meeting (not in the next three weeks, though, as both @mnonnenmacher and I will be on vacation).

**Review by sschuberth — APPROVED** [review]

**Pull request merged**
